test/types.ts

```diff
@@ -4,148 +4,86 @@
  */
 
 import {
-  expectTypeOf,
-} from 'expect-type';
+  expectType,
+} from 'tsd';
```

Comment:
I'm the author of expect-type and not tsd, so I might be biased (although both are free!) - but I think a switch to tsd should be done _very_ carefully if dealing with generics in any way because of this bug: https://github.com/SamVerschueren/tsd/issues/142

IMO it's not worth the risk. It's a serious bug for a type assertion library because it'll tell you everything is fine even when it's not. expect-type doesn't have that problem (and I think the assertions are a little more readable).

Reply:
Fair enough – re-added `expect-type`.